### ISSUE TITLE
modernize-use-ranges false positive for std::fill on std::vector<bool>

### ISSUE THREAD

**Issue by fmun** [llvm/llvm-project#195296]
clang-tidy version 22.1.3 for C++20 recommends to use a ranges algorithm to fill a `std::vector<bool>`:

```
#include <vector>

int main() {
    std::vector<bool> v(10);

    // Fill the entire range with true
    std::fill(v.begin(), v.end(), true); // warning: use a ranges version of this algorithm [modernize-use-ranges]
}
```

The `std::ranges::fill` algorithm, however, cannot be used with `std::vector<bool>` in C++20.

https://godbolt.org/z/evj8r8azW

**Comment by zeyi2** [comment]
With `-fix` it will give invalid-fixits, so I'll update the labels and parent issues

**Comment by Supercool1604** [comment]
Hi,

Can i please be assigned to this issue?
